packages/block-library/src/navigation/style.scss

```diff
@@ -130,6 +130,7 @@
 		// Don't take up space when the menu is collapsed.
 		width: 0;
 		height: 0;
+		overflow: hidden; // Overflow is necessary to set, otherwise submenu items will take up space.
```

Comment:
This overflow rule I believe I had in the past, and then removed it. In this PR it appears to work as intended, with no ill side effects. But I would appreciate thorough testing of submenus, with and without click, and with and without keyboard usage. Just to make sure I'm not missing anything.